src/writer.rs

```diff
@@ -329,3 +392,39 @@ where
     writer.write_all(&*bytes)?;
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_writer_path() {
+        let w = WriterOptions::new("~/image");
+        assert_eq!(w.header_path, PathBuf::from("~/image.nii"));
+
+        for (p, nii, niigz) in &[
+            ("~/image", "~/image.nii", "~/image.nii.gz"),
+            ("~/image.nii", "~/image.nii", "~/image.nii.gz"),
+            ("~/image.nii.gz", "~/image.nii", "~/image.nii.gz"),
+            ("~/image.hdr", "~/image.hdr", "~/image.hdr.gz"),
```

Comment:
Do these tests demonstrate that enabling compression on an options with a header file path `"~/image.hdr"` would convert it to `"~/image.hdr.gz"`, and therefore compress the header? I could be mistaken, but is there interest in compressing both header and volume files? The reader only supports this for flexibility, but my impression is that header files are more often left uncompressed even if the respective .img is compressed.

Reply:
I removed the support for writing compressed headers because, as you wrote, it's mostly useless. Asking for

    WriterOptions::new("brain.nii.gz").write_header_file(true).write_nifti(&data).unwrap();
    or
    WriterOptions::new("brain.nii").write_header_file(true).compress(true).write_nifti(&data).unwrap();

would write `brain.nii.gz` and `brain.hdr`. However, what if the user asks

    WriterOptions::new("brain.hdr.gz").write_header_file(true).write_nifti(&data).unwrap();

With my current version, we write `brain.nii.gz` and `brain.hdr`... I'm pretty sure that the user wants`brain.hdr.gz`. Of course, this broke `write_hdr_standard` which write a hdr and hdr.gz file, then tries to read the same path.

What's your opinion on this? Should we support writing compressed headers only when the user specifically asks for it?

Reply:
Granted, if the user wrote "whatever.hdr.gz", then they are clearly expecting the header to be compressed. I think we should honor that, even if this means adding another property to our writer options.

In any other case (such as stating "whatever.nii[.gz]"), I say we always write the header uncompressed.